docs/libcurl/opts/CURLOPT_ECH.3

```diff
@@ -0,0 +1,74 @@
+.\" **************************************************************************
+.\" *                                  _   _ ____  _
+.\" *  Project                     ___| | | |  _ \| |
+.\" *                             / __| | | | |_) | |
+.\" *                            | (__| |_| |  _ <| |___
+.\" *                             \___|\___/|_| \_\_____|
+.\" *
+.\" * Copyright (C) Daniel Stenberg, <daniel@haxx.se>, et al.
+.\" *
+.\" * This software is licensed as described in the file COPYING, which
+.\" * you should have received as part of this distribution. The terms
+.\" * are also available at https://curl.se/docs/copyright.html.
+.\" *
+.\" * You may opt to use, copy, modify, merge, publish, distribute and/or sell
+.\" * copies of the Software, and permit persons to whom the Software is
+.\" * furnished to do so, under the terms of the COPYING file.
+.\" *
+.\" * This software is distributed on an "AS IS" basis, WITHOUT WARRANTY OF ANY
+.\" * KIND, either express or implied.
+.\" *
+.\" * SPDX-License-Identifier: curl
+.\" *
+.\" **************************************************************************
+.\"
+.TH CURLOPT_ECH 3 "30 Sep 2023" libcurl libcurl
+.SH NAME
+CURLOPT_ECH \- control configuration for Encrypted Client Hello
+.SH SYNOPSIS
+.nf
+#include <curl/curl.h>
+
+CURLcode curl_easy_setopt(CURL *handle, CURLOPT_ECH, char *config);
+.fi
+.SH DESCRIPTION
+Pass a string that specifies configuration details for ECH.
+In all cases, if ECH is attempted, it may fail for various reasons.
+The keywords supported are:
+.IP false
+Turns off ECH.
+.IP grease
+Instructs client to emit a GREASE'd ECH extension.
+.IP true
+Instructs client to attempt ECH, if possible, but to not fail if attempting ECH is not possible.
+.IP hard
+Instructs client to attempt ECH and fail if if attempting ECH is not possible.
+.IP ecl:<base64-value>
+If the string starts with "ecl:" then the remainder of the string should be a base64-encoded
+ECHConfigList that will be used for ECH rather than attempting to download such a value from
+the DNS.
+.IP pn:<name>
+If the string starts with "pn:" then the remainder of the string should be a DNS/host name
+that will be used to over-ride the public_name field of the ECHConfigList that will be used
+for ECH.
+.SH DEFAULT
+There is no default value, meaning ECH is disabled if no
+configuration is supplied.
+.SH PROTOCOLS
+TLS, and requires TLS1.3.
+.SH EXAMPLE
+.nf
+CURL *curl = curl_easy_init();
+
+const char *config ="ecl:AED+DQA87wAgACB/RuzUCsW3uBbSFI7mzD63TUXpI8sGDTnFTbFCDpa+CAAEAAEAAQANY292ZXIuZGVmby5pZQAA";
+if(curl) {
+  curl_easy_setopt(curl, CURLOPT_ECH, config);
+  curl_easy_perform(curl);
+}
+.fi
+.SH AVAILABILITY
+Added in 8.4.0
```

Comment:
This seems likely to happen in 8.6.0 as things look.

Reply:
done (for now:-)